### ISSUE TITLE
"command not found" given for calls to executables located in virtualenv `bin/` on python 2.7

### ISSUE THREAD

**Issue by tuukkamustonen** [pyinvoke/invoke#479]
Noticed this while testing fabric 2.

With latest commit https://github.com/fabric/fabric/tree/ecb588b788cfea5d2becc6957ad12e8c78042a38:

    mkvirtualenv testenv
    pip install https://github.com/fabric/fabric/archive/v2.zip

Create `fabfile.py`:

```python
from invoke import task

@task
def foo(c):
    c.run('pip')
```

`fab foo` yields:

    /bin/bash: pip: command not found

Both python 2.7 & 3.5 give the same result.

Rename `fabfile.py` as `tasks.py` and run `inv foo` instead. Result:

- Python 2.7 fails as earlier
- Python 3.5 works now

Runnnig invoke 0.21.0.

`pty` has no impact on result.

Running `c.run('ls')` or similar works just fine - it's just commands installed inside virtualenv's `bin` that seem to fail.

Don't have other python 3 versions at hand to test now...

**Comment by bitprophet** [comment]
Hrm, I can't recreate this under Python 2.7. Last I checked (I've been confused on this in the past...) virtualenvs rely solely on `$PATH` so as long as the inner env preserves it, it ought to "find" the right pip/python/etc no problem. Seems the case here on latest Invoke master, adding a new tiny task to its internal management `tasks.py`:

![image](https://user-images.githubusercontent.com/6088/31416028-6fc9cca0-addb-11e7-8757-ca2bf52320a4.png)

Identical result under 0.21.0, as well as under Python 3(.5).

----

So I'd start by introspecting the internal shell's env, as I did above, in case something wacky is going on there. Perhaps dotfiles are coming into play? In my case, while Invoke defaults to bash right now, my personal shell is zsh, so I have no bash dotfiles floating around.
